### ISSUE TITLE
Cannot have more than one autoconfigured CC toolchain

### ISSUE THREAD

**Issue by mboes** [bazelbuild/bazel#6696]
### Description of the problem

Bazel currently implicitly adds a call to `tools/cpp/cc_configure:cc_configure()` at the end of every `WORKSPACE` file. This autoconfigures a CC toolchain. We can override the autoconfiguration to some extent, by creating our own wrapper to `cc_autoconf_impl()`, in particular to supply a non-empty `overriden_tools` argument. But the resulting toolchain ultimately _must_ be defined in an external repository named `@local_cc_config`. We cannot autoconfigure a toolchain definition in some other repository with some other name.

If we could use a different repository name, then we could have both the default autoconfigured CC toolchain and any toolchain with overridden tools coexist. Why would this be useful? Because my `overriden_tools` aren't always available in all environments I'm running builds on. If I can define my own autoconfigured toolchain in say `@foo_cc_config`, then when this toolchain is not available I can force Bazel to use `@local_cc_config` using `--extra_toolchains` on the command-line.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

In any project with the following `WORKSPACE` file:

```python
workspace(name = "test")

http_archive(
    name = "io_tweag_rules_nixpkgs",
    strip_prefix = "rules_nixpkgs-0.4.1",
    urls = ["https://github.com/tweag/rules_nixpkgs/archive/v0.4.1.tar.gz"],
    sha256 = "e08bfff0e3413cae8549df72e3fce36f7b0e2369e864dfe41d3307ef100500f8",
)

load(
    "@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl",
    "nixpkgs_git_repository",
    "nixpkgs_cc_configure",
)

nixpkgs_git_repository(
    name = "nixpkgs",
    remote = "https://github.com/NixOS/nixpkgs",
    revision = "18.09",
    sha256 = "6451af4083485e13daa427f745cbf859bc23cb8b70454c017887c006a13bd65e",
)

nixpkgs_package(
  name = "ghc",
  attribute_path = "haskell.compiler.ghc843",
  repository = "@nixpkgs",
)

nixpkgs_cc_configure(repository = "@nixpkgs")
```

The above works, but that's because `nixpkgs_cc_configure()` hardcodes defining a `@local_cc_config` repository under the hood. If you change the source code of `nixpkgs_cc_configure()` to a different hardcoded name, then we get the following error:

```python
ERROR: /home/mboes/.cache/bazel/_bazel_mboes/ca2b5791ec7fc9a5f1d8b65c31efa007/external/foo_config_cc/BUILD:71:1: target '@bazel_tools//platforms:autoconfigured' is not visible from target '@foo_config_cc//:cc-toolchain-k8'. Check the visibility declaration of the former target if you think the dependency is legitimate
```

So perhaps the fix is simply to let the autoconfigured platform be visible from any external repository, no matter its name.

### What operating system are you running Bazel on?

NixOS

### What's the output of `bazel info release`?

`Build label: 0.18.0- (@non-git)`

### If `bazel info release` returns "development version" or "(@non-git)", tell us how you built Bazel.

From Nixpkgs

###  Have you found anything relevant by searching the web?

No. Only the source code for Envoy, which similarly defines `@local_cc_config`, rather than using a different name. See https://github.com/envoyproxy/envoy/blob/master/bazel/cc_configure.bzl#L123.

**Comment by katre** [comment]
This isn't really something that can be handled by toolchain resolution. Re-assigning to @hlopko because he understands the CC autoconfiguration better than I do.

**Comment by hlopko** [comment]
So is the only problem the autoconfigured target visibility? Because that is gone now: https://github.com/bazelbuild/bazel/pull/6705 and is already released in 0.21.

I'm fine with adding a argument to cc_configure specifying the name of the repo, it's actually a cool idea. Will you give it a shot Mathieu?

Thanks!

**Comment by jin** [comment]
@scentini could you add a [priority](https://www.bazel.build/maintainers-guide.html#priority) to this issue, please?
